R/particle_filter_state_nested.R

```diff
@@ -83,31 +84,55 @@ particle_filter_state_nested <- R6::R6Class(
       }
 
       if (!is.null(initial)) {
-        initial_data <- initial(model$info(), n_particles, pars)
-        if (is.list(initial_data)) {
-          steps <- particle_steps(steps, initial_data$step)
-          model$set_state(initial_data$state, initial_data$step)
+        initial_data <- Map(initial,
+          info = model$info(),
+          pars = pars,
+          MoreArgs = list(n_particles = n_particles)
+        )
+        if (!is.null(unlist(initial_data))) {
+          if (any(c(c("step", "state") %in% names(initial_data[[1]])))) {
```

Comment:
Generally we probably want to not just check the first element but all. So perhaps something like:

```
initial_step <- lapply(initial_data, "[[", "step")
initial_state <- lapply(initial_data, "[[", "state")
```

then work with these lists

Reply:
I don't know if the current setting allows one element to have 'step' or 'state' and not the other. The `initial` function (at least not in tests or carehomes) doesn't differentiate between what (step/state/etc) to return for populations, just the values they take.

Reply:
yes, either could be set but if any parameter sets one all must set it